File: ghost-env/src/recording/export.ts
import type { CallRecord } from "./recorder.js";

export function exportRecordingJSON(calls: CallRecord[]): string {
  return JSON.stringify(calls, null, 2);
}

export function exportRecordingMarkdown(calls: CallRecord[]): string {
  const lines = ["# ghost-env recording", ""];
  for (const c of calls) {
    lines.push(`## ${c.provider} ${c.method} ${c.url}`);
    lines.push(`- status: ${c.responseStatus}`);
    lines.push(`- durationMs: ${c.durationMs}`);
    lines.push("");
  }
  return lines.join("\n");
}

export function exportHAR(calls: CallRecord[], title = "ghost-env"): string {
  const entries = calls.map((c, i) => ({
    startedDateTime: new Date().toISOString(),
    time: c.durationMs,
    request: {
      method: c.method,
      url: c.url,
      headers: [],
    },
    response: {
      status: c.responseStatus,
      content: { text: typeof c.responseBody === "string" ? c.responseBody : JSON.stringify(c.responseBody) },
    },
    _provider: c.provider,
    _id: c.id,
  }));
  return JSON.stringify({ log: { version: "1.2", creator: { name: title }, entries } }, null, 2);
}

File: ghost-env/python/src/ghost_env/export.py
from __future__ import annotations

import json
from typing import Any

from ghost_env.recorder import CallRecord


def export_recording_json(calls: list[CallRecord]) -> str:
    return json.dumps([c.__dict__ for c in calls], indent=2)


def export_recording_markdown(calls: list[CallRecord]) -> str:
    lines = ["# ghost-env recording", ""]
    for c in calls:
        lines.append(f"## {c.provider} {c.method} {c.url}")
        lines.append(f"- status: {c.response_status}")
        lines.append("")
    return "\n".join(lines)


def export_har(calls: list[CallRecord], title: str = "ghost-env") -> str:
    entries = []
    for c in calls:
        entries.append(
            {
                "request": {"method": c.method, "url": c.url},
                "response": {"status": c.response_status, "content": {"text": str(c.response_body)}},
            }
        )
    return json.dumps({"log": {"version": "1.2", "creator": {"name": title}, "entries": entries}}, indent=2)
